lib/integrations/awsoidc/list_ec2ice_test.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/google/go-cmp/cmp"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+type mockListEC2ICEClient struct {
+	pageSize  int
+	accountID string
+	ec2ICEs   []ec2Types.Ec2InstanceConnectEndpoint
+}
+
+// Returns information about ec2 instances.
+// This API supports pagination.
+func (m mockListEC2ICEClient) DescribeInstanceConnectEndpoints(ctx context.Context, params *ec2.DescribeInstanceConnectEndpointsInput, optFns ...func(*ec2.Options)) (*ec2.DescribeInstanceConnectEndpointsOutput, error) {
+	requestedPage := 1
+
+	totalEndpoints := len(m.ec2ICEs)
+
+	if params.NextToken != nil {
+		currentMarker, err := strconv.Atoi(*params.NextToken)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		requestedPage = currentMarker
+	}
+
+	sliceStart := m.pageSize * (requestedPage - 1)
+	sliceEnd := m.pageSize * requestedPage
+	if sliceEnd > totalEndpoints {
+		sliceEnd = totalEndpoints
+	}
+
+	ret := &ec2.DescribeInstanceConnectEndpointsOutput{
+		InstanceConnectEndpoints: m.ec2ICEs[sliceStart:sliceEnd],
+	}
+
+	if sliceEnd < totalEndpoints {
+		nextToken := strconv.Itoa(requestedPage + 1)
+		ret.NextToken = stringPointer(nextToken)
```

Comment:
calling `stringPointer` seems an unnecessary indirection. This line could just be `ret.NextToken = &nextToken`. `stringPointer()` likely exists to be able to take the address of a string literal, but since you have a variable here you shouldn't need to call it.

Reply:
Yes, I should just use `&nextToken`